src/storage/slab/item.c

```diff
@@ -128,6 +128,12 @@ _item_link(struct item *it)
     PERSLAB_INCR_N(it->id, item_val_byte, it->vlen);
 }
 
+bool
+item_will_fit(const struct item *it, uint32_t len)
+{
+    return slab_id(item_size(it)) == item_slabid(it->klen, it->vlen + len, it->olen);
```

Comment:
I think it's cheap if we:
- inline this function;
- make (slab) profile visible (so we can define an inline function in the header);
- implement as `return (profile[it->id] <= item_size(it) + len);`

Mostly this is cheaper because `item_slabid` does a binary search through `profile` array.

Reply:
Also, maybe changing `len` something like `delta` or `increment` will make the purpose of the function even more obvious?